### ISSUE TITLE
Craft 4 Support

### ISSUE THREAD

**Issue by andypullen** [alpshq/craft-share-previews#12]
Hello,
Are you planning on supporting Craft 4?
Thanks

**Comment by jakubjo** [comment]
Hi @andypullen -- yes! Support for Craft 4.0 is comming soon. I'll keep this issue open until its ready.

**Comment by arifje** [comment]
Any update on this? :-)

**Comment by andypullen** [comment]
Hi @jakubjo, do you have a timeline for this? Appreciate it!

**Comment by jakubjo** [comment]
Hi @andypullen & @arifje! 

I've tagged `2.0.0` with support for Craft 4 -- please Excuse the delay in adding support for craft version 4.
If you encounter any issue with Craft 4, please report them either here, or create a new issue.

Best regards,
Jakub

**Comment by andypullen** [comment]
Using this plugin along with SEOmatic in Craft 4 no longer works. I still have `{entry.getSharePreviewUrl()}` in the meta image URL field results in this exception:
```
Twig\Error\RuntimeError: Calling unknown method: craft\models\FieldLayout::getFields() in /var/www/html/templates/campuses/podcasts/episodes.twig:154
Stack trace:
#0 /var/www/html/storage/runtime/compiled_templates/cc/cc369e2073a4970b8f6b2e10236be40b3a57b90f04da70f6e19bf0a61d63506f.php(202): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(craft\elements\Entry), 'getSharePreview...', Array, 'method')
#1 /var/www/html/vendor/twig/twig/src/Template.php(171): __TwigTemplate_3ed631dfe3a8113816b87e3bc0c1a4dce37c3003156581bb352237149de581f8->block_body(Array, Array)
#2 /var/www/html/storage/runtime/compiled_templates/ab/abdbd759becbb758d457f7330dc3f7f491233afe42d46df6ba55605974f7f5fd.php(99): Twig\Template->displayBlock('body', Array, Array)
#3 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_4a748a0315c7092e7f25475498438e94b7a69ca932f62b159f522402133454d3->doDisplay(Array, Array)
#4 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#5 /var/www/html/storage/runtime/compiled_templates/cc/cc369e2073a4970b8f6b2e10236be40b3a57b90f04da70f6e19bf0a61d63506f.php(175): Twig\Template->display(Array, Array)
#6 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_3ed631dfe3a8113816b87e3bc0c1a4dce37c3003156581bb352237149de581f8->doDisplay(Array, Array)
#7 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#8 /var/www/html/storage/runtime/compiled_templates/21/21ec5ccc92ab8f095d5c207db8aa9a8e4185b7c57c6318dda7a58a0e21b98d4a.php(127): Twig\Template->display(Array)
#9 /var/www/html/vendor/twig/twig/src/Template.php(394): __TwigTemplate_e7be058938bbe3b08fa40304634f3ba5d96ac6f29eaf585c761e15a544a4a3ae->doDisplay(Array, Array)
#10 /var/www/html/vendor/twig/twig/src/Template.php(367): Twig\Template->displayWithErrorHandling(Array, Array)
#11 /var/www/html/vendor/twig/twig/src/Template.php(379): Twig\Template->display(Array)
#12 /var/www/html/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\Template->render(Array, Array)
#13 /var/www/html/vendor/twig/twig/src/Environment.php(277): Twig\TemplateWrapper->render(Array)
#14 /var/www/html/vendor/craftcms/cms/src/web/View.php(451): Twig\Environment->render('campuses/podcas...', Array)
#15 /var/www/html/vendor/craftcms/cms/src/web/View.php(504): craft\web\View->renderTemplate('campuses/podcas...', Array)
#16 /var/www/html/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(56): craft\web\View->renderPageTemplate('campuses/podcas...', Array, 'site')
#17 /var/www/html/vendor/yiisoft/yii2/web/Response.php(1098): craft\web\TemplateResponseFormatter->format(Object(craft\web\Response))
#18 /var/www/html/vendor/craftcms/cms/src/web/Response.php(286): yii\web\Response->prepare()
#19 /var/www/html/vendor/yiisoft/yii2/web/Response.php(339): craft\web\Response->prepare()
#20 /var/www/html/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#21 /var/www/html/web/index.php(21): yii\base\Application->run()
#22 {main}
```

**Comment by jakubjo** [comment]
@andypullen thanks for reporting! 

It should be fixed in `2.1.1` -- can you verify by running `composer update`?

**Comment by andypullen** [comment]
Looks to be working now. Thank you for the quick response!
